build/charts/antrea-ui/README.md

```diff
@@ -27,14 +27,15 @@ Kubernetes: `>= 1.16.0-0`
 | frontend.resources | object | `{}` | Resource requests and limits for the frontend container. |
-| https | object | `{"auto":{"commonName":"localhost","daysValid":365,"dnsNames":[],"ipAddresses":[]},"enable":false,"method":"auto","user":{"cert":"","key":""},"userCA":{"cert":"","commonName":"localhost","daysValid":365,"dnsNames":[],"ipAddresses":[],"key":""}}` | HTTPS configuration for the Antrea UI. |
+| https | object | `{"auto":{"commonName":"localhost","daysValid":365,"dnsNames":[],"ipAddresses":[]},"enable":false,"method":"auto","secret":{"secretName":"antrea-ui-tls"},"user":{"cert":"","key":""},"userCA":{"cert":"","commonName":"localhost","daysValid":365,"dnsNames":[],"ipAddresses":[],"key":""}}` | HTTPS configuration for the Antrea UI. |
 | https.auto | object | `{"commonName":"localhost","daysValid":365,"dnsNames":[],"ipAddresses":[]}` | Configure automatic TLS certificate generation with Helm. |
 | https.auto.commonName | string | `"localhost"` | Common name to use in the certificate. |
 | https.auto.daysValid | int | `365` | Number of days for which the certificate will be valid. There is no automatic rotation with this method. |
 | https.auto.dnsNames | list | `[]` | DNS names to use in the certificate. |
 | https.auto.ipAddresses | list | `[]` | IP addresses to use in the certificate. |
 | https.enable | bool | `false` | Enable HTTPS (only) for accessing the web UI. |
-| https.method | string | `"auto"` | Method for generating the TLS certificate for the web server. We support "auto", "user", and "userCA". With "auto", Helm will generate a new self-signed certificate every time the template function is executed. With "user", the user is responsible for providing a certificate and key, which will be used directly. With "userCA", the user is responsible for providing a CA certificate and key, which will be used to generate a signed certificate to be used by the web server. |
+| https.method | string | `"auto"` | Method for generating the TLS certificate for the web server. We support "auto", "user", and "userCA". With "auto", Helm will generate a new self-signed certificate every time the template function is executed. With "user", the user is responsible for providing a certificate and key, which will be used directly. With "userCA", the user is responsible for providing a CA certificate and key, which will be used to generate a signed certificate to be used by the web server. With "secret", the user is responsible for providing a secret of type kubernetes.io/tls, in the Namespace of the release. The secret must include the tls.crt and tls.key data fields. |
+| https.secret.secretName | string | `"antrea-ui-tls"` | Name of the secret containing the PEM data for the certificate and private key to use. Secret must be of type kubernetes.io/tls. The typical use case is a secret generated by cert-manager. The secret must exist in the Namespace of the Helm release (typically, kube-system). |
```

Comment:
Is it simpler to let the secret to store the PEM data for antrea-ui regardless of how it is generated? Two pros I think of:
1. User could always check the secret for the PEM data.
2. The secret name in the manifest is unified.

Reply:
I may be misunderstanding your comment, but I think you are asking why we don't use a single secret (`antrea-ui-tls`) for all cases / methods?

The reason why I use 2 different secrets is that one is essentially owner by the user (`antrea-ui-tls` by default, but user can specify another name), while the other is owned by Helm (`antrea-ui-tls-helm`). I worry that if I use the same secret, then it could lead to conflicts if the user tries different HTTPS "methods" (with `helm upgrade`). For example, if we use a single unified secret, it's not clear to me when we should let Helm create the secret? If Helm attempts to create a secret that already exists, then Helm will fail, yet we want Helm to create the secret for `"auto"` / `"user"` / `"userCA"`. On the other hand, if Helm always creates the secret (including when using cert-manager), then the user has to ensure that the secret doesn't exist before hand (or Helm will fail). If the user does `helm uninstall`, the secret will also be deleted, but may be re-created by cert-manager soon after. Then a subsequent `helm install` would fail.

I was considering using the `lookup` Helm function and potentially checking if the secret already exists, but it didn't seem clean and it only works when Helm is used with a valid K8s cluster (not when generating manifests).

What do you think? I would also like to have a unified approach, but couldn't think of any good solution?

Reply:
Yes, that's my question. But I didn't know it's because of technical difficultyies. I don't know other solutions so I'm fine with it.